File: cdm/java/mil/tatrc/physiology/datamodel/properties/SEScalarNegative1To1.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package mil.tatrc.physiology.datamodel.properties;

import com.kitware.physiology.cdm.Properties.ScalarNegative1To1Data;

import mil.tatrc.physiology.utilities.Log;

/**
 * @author abray
 * 0To1 Scalar Class, will enforce that value is between 0 and 1 (inclusive)
 */

public class SEScalarNegative1To1 extends SEScalar
{
  public SEScalarNegative1To1()
  {
    super();
  }
  
  /**
   * @param value
   * @param unit - specific any valid standard unit abbreviation
   * http://www.bipm.org/en/si/ for this quantity type
   */
  public SEScalarNegative1To1(double value)
  {
    this();
    this.setValue(value, "");
  }
  
  public static void load(ScalarNegative1To1Data src, SEScalarNegative1To1 dst)
  {
    SEScalar.load(src.getScalarNegative1To1(),dst);
  }
  public static ScalarNegative1To1Data unload(SEScalarNegative1To1 src)
  {
    if(!src.isValid())
      return null;
    ScalarNegative1To1Data.Builder dst = ScalarNegative1To1Data.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  protected static void unload(SEScalarNegative1To1 src, ScalarNegative1To1Data.Builder dst)
  {
    SEScalar.unload(src,dst.getScalarNegative1To1Builder());
  }
 
  public boolean validUnit(String unit)
  {
    if(unit==null||unit.isEmpty()||unit.compareTo("unitless") == 0)
      return true;
    return false;
  }
  
  public void setValue(double d)
  {
    if(d>1)
    {
      d=1;
      Log.warn("Attempting to set [-1,1] Scalar to greater than 1, bounding to 1");
    }
    else if(d<-1)
    {
      d=-1;
      Log.warn("Attempting to set [-1,1] Scalar to less than -1, bounding to -1");
    }
    this.value = d;
  }
}

File: cdm/cpp/properties/SEScalarNegative1To1.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "properties/SEScalarNegative1To1.h"

SEScalarNegative1To1::SEScalarNegative1To1() : SEScalar()
{

}

void SEScalarNegative1To1::SetValue(double d)
{ 
  if (d > 1 || d < -1)
    throw CommonDataModelException("SEScalarNegative1To1 must be between [-1,1]");
  SEScalar::SetValue(d);
}

void SEScalarNegative1To1::SetValue(double d, const NoUnit& unitless)
{ 
  SEScalarNegative1To1::SetValue(d);
}

void SEScalarNegative1To1::Load(const cdm::ScalarNegative1To1Data& src, SEScalarNegative1To1& dst)
{
  SEScalarNegative1To1::Serialize(src, dst);
}
void SEScalarNegative1To1::Serialize(const cdm::ScalarNegative1To1Data& src, SEScalarNegative1To1& dst)
{
  SEScalar::Serialize(src.scalarnegative1to1(), dst);
}

cdm::ScalarNegative1To1Data* SEScalarNegative1To1::Unload(const SEScalarNegative1To1& src)
{
  if (!src.IsValid())
    return nullptr;
  cdm::ScalarNegative1To1Data* dst = new cdm::ScalarNegative1To1Data();
  Serialize(src, *dst);
  return dst;
}
void SEScalarNegative1To1::Serialize(const SEScalarNegative1To1& src, cdm::ScalarNegative1To1Data& dst)
{
  SEScalar::Serialize(src, *dst.mutable_scalarnegative1to1());
}                                       